src/reducetensor.cpp

```diff
@@ -552,6 +557,12 @@ void ReduceTensorDescriptor::ReduceTensor(const Handle& handle,
     // to remove the warning from clang-tidy checking
     param += " -DMIOPEN_USE_FP32=0 -DMIOPEN_USE_FP16=0 ";
 
+#if WORKAROUND_MIOPEN_ISSUE_557
+    const auto device_name = handle.GetDeviceName();
+    if ( StartsWith(device_name, "gfx1030") )
```

Comment:
```suggestion
    if ( StartsWith(handle.GetDeviceName(), "gfx10") )
```
Why:
- I think that old intrinsics do not work for all gfx10 targets.
- We do not need `device_name` after this. Let's use the expressiveness of the C++ language.

Reply:
Corrected. Thanks